File: backend/app/schemas/sale.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class SaleItemCreate(BaseModel):
    medicine_id: int
    quantity: int


class SaleCreate(BaseModel):
    customer_id: int
    payment_method: str
    discount: float = 0
    items: list[SaleItemCreate]


class SaleItemResponse(BaseModel):
    medicine_id: int
    quantity: int
    unit_price: float
    total_price: float

    model_config = ConfigDict(from_attributes=True)


class SaleResponse(BaseModel):
    id: int
    invoice_number: str
    customer_id: int
    subtotal: float
    discount: float
    tax: float
    grand_total: float
    payment_method: str
    sale_date: datetime

    items: list[SaleItemResponse]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/sale.ts
export type SaleItem = {
  medicine_id: number;
  quantity: number;
  unit_price: number;
  total_price: number;
};

export type Sale = {
  id: number;
  invoice_number: string;
  customer_id: number;
  subtotal: number;
  discount: number;
  tax: number;
  grand_total: number;
  payment_method: string;
  sale_date: string;
  items: SaleItem[];
};